Origin: https://mail.openvswitch.org/pipermail/ovs-dev/2024-September/417066.html

diff --git a/controller/route-exchange-netlink-private.h b/controller/route-exchange-netlink-private.h
index 4c2559895..13338366a 100644
--- a/controller/route-exchange-netlink-private.h
+++ b/controller/route-exchange-netlink-private.h
@@ -127,6 +127,10 @@ route_table_parse(struct ofpbuf *buf, struct route_table_msg *change)
             change->relevant = false;
         }
 
+        if (rtm->rtm_protocol != RTPROT_BOOT) {
+            change->relevant = false;
+        }
+
         table_id = rtm->rtm_table;
         if (attrs[RTA_TABLE]) {
             table_id = nl_attr_get_u32(attrs[RTA_TABLE]);
diff --git a/controller/route-exchange-netlink.c b/controller/route-exchange-netlink.c
index 707676f33..e04a35520 100644
--- a/controller/route-exchange-netlink.c
+++ b/controller/route-exchange-netlink.c
@@ -216,6 +216,10 @@ handle_route_msg_delete_host_routes(struct route_table_msg *msg, void *data)
     struct host_route_node *hr;
     int err;
 
+    if (!msg->relevant) {
+        return;
+    }
+
     uint32_t hash = host_route_hash(&rd->rta_dst);
     HMAP_FOR_EACH_WITH_HASH (hr, hmap_node, hash, host_routes) {
         if (ipv6_addr_equals(&hr->addr, &rd->rta_dst)) {
